paf-mvp-operator-express/src/operator-node.ts

```diff
@@ -765,14 +765,26 @@ export class OperatorNode extends Node {
 
   catchErrorsWithRedirectToReferer =
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    (error: unknown, req: Request, res: Response, next: NextFunction) => {
+    (err: unknown, req: Request, res: Response, next: NextFunction) => {
       const { endPointName } = this.getRequestConfig(req);
-      this.logger.Error(endPointName, error);
-      const nodeError: NodeError = {
-        type: NodeErrorType.UNKNOWN_ERROR,
-        details: '',
-      };
-      this.redirectWithError(res, req.header('referer'), 500, nodeError);
+      this.logger.Error(endPointName, err);
+
+      // FIXME[errors] this whole method must be merged with catchErrors to avoid duplication like this
+      if ((err as Error).message === 'Response timeout') {
+        const error: NodeError = {
+          type: NodeErrorType.RESPONSE_TIMEOUT,
+          details: (err as Error).message,
+        };
+        this.redirectWithError(res, req.header('referer'), 504, error);
+      } else if ((err as NodeError).type) {
+        const nodeError: NodeError = {
+          type: NodeErrorType.UNKNOWN_ERROR,
+          details: '',
+        };
+        this.redirectWithError(res, req.header('referer'), 500, nodeError);
+      }
+
+      next();
```

Comment:
I think that the `next()` call here is the good one but it may require a bit of comment because there are implicit things that are going on here.

The `next()` call in this way is meaningful in an error handler when there is no more error handler after. If there is another error handler after, it should be `next(err)` so that the next handler isn't skipped. It means that the error handler isn't agnostic to its execution context and needs to know if there is another error handler after or not.

To avoid any bad copy-past or bad changes, can we annotate this line with a comment stating that we have finished handling the error at this point, that there is no more error handler and we call `next()` to pass the requests to a standard handler if there is one.